src/scanner/scanner.py

```diff
@@ -0,0 +1,328 @@
+                    sum(abs(curve.richards - curve.casos_cum))
+                    / max(curve.casos_cum)
+                ),
+                "ep_time": comp_duration(curve),
+            }
+        )
+
+    def _parse_results(self) -> pd.DataFrame:
+        data = {
+            "geocode": [],
+            "year": [],
+            "muni_name": [],
+            "peak_week": [],
+            "beta": [],
+            "gamma": [],
+            "R0": [],
+            "total_cases": [],
+            "alpha": [],
+            "sum_res": [],
+            "ep_ini": [],
+            "ep_end": [],
+            "ep_dur": [],
+        }
+
+        for gc, curve in self.curves.items():
+            for c in curve:
+                data["geocode"].append(gc)
+                data["muni_name"].append(get_municipality_name(gc))
+                data["year"].append(c["year"])
+                params = [p["params"] for p in self.results[gc]][0]
+                sir_params = [p["sir_pars"] for p in self.results[gc]][0]
+                data["peak_week"].append(params["tp1"])
+                data["total_cases"].append(params["L1"])
+                data["alpha"].append(params["a1"])
+                data["beta"].append(sir_params["beta"])
+                data["gamma"].append(sir_params["gamma"])
+                data["R0"].append(sir_params["R0"])
+                data["sum_res"].append(c["sum_res"])
+
+                ep_duration = c["ep_time"]
+                data["ep_ini"].append(ep_duration["ini"])
+                data["ep_end"].append(ep_duration["end"])
+                data["ep_dur"].append(ep_duration["dur"])
+
+        return pd.DataFrame(data)
+
+    async def _scan(self, geocode):
+        df = self._filter_city(geocode)
+        df = df.assign(year=[i.year for i in df.index])
+
+        dfy = df[df.year == self.year]
+        window = int(max([str(x)[-2:] for x in dfy.SE]))
+        has_transmission = dfy.transmissao.sum() > 3
+
+        if not has_transmission:
+            if self.verbose:
+                logger.info(
+                    f"""
+                    There were less than 3 weeks with Rt>1
+                    in {geocode}.\nSkipping analysis
+                    """
+                )
+            return
+
+        out, curve = otim(
+            df[["casos", "casos_cum"]].iloc[0:window],  # NOQA E203
+            0,
+            window,
+        )
+
+        self._save_results(geocode, out, curve)
+
+        if out.success:
+            if self.verbose:
+                logger.info(
+                    f"""
+                        R0: {
+                        self.results[geocode][-1]['sir_pars']['R0']
+                    }
+                    """
+                )
+
+    async def _scan_all_geocodes(self):
+        tasks = [
+            self._scan(geocode)
+            for geocode in self.data.municipio_geocodigo.unique()
+        ]
+        await asyncio.gather(*tasks)
+
+    def _to_duckdb(self, output_dir: str):
```

Comment:
@fccoelho @esloch this method does the table overriding according to the year scanned, the table's name is the UF.